shaders/CurrentsParticles.gdshader

```diff
@@ -0,0 +1,122 @@
+shader_type particles;
+render_mode disable_velocity;
+
+uniform vec3 emission_box_extents;
+uniform vec4 colorValue : source_color;
+uniform sampler2D alphaCurve : repeat_disable;
+
+uniform float gameTime;
+
+uniform sampler3D noiseDisturbancesX;
+uniform sampler3D noiseDisturbancesY;
+
+uniform sampler3D noiseCurrentsX;
+uniform sampler3D noiseCurrentsY;
+
+uniform float speed;
+uniform float chaoticness;
+uniform float scale;
+
+// The following constants should be the same as in FluidCurrentsSystem.cs
+const float disturbanceTimescale = 1.0f;
+const float currentsTimescale = 1.0f / 500.0f;
+const float currentsStretchingMultiplier = 1.0f / 10.0f;
+const float disturbanceToCurrentsRatio = 0.15f;
+const float minCurrentIntensity = 0.4f;
+const float positionScaling = 0.9f;
+
+// Generates a random float and modifies the seed
+float rand_from_seed(inout uint seed) {
+    int k;
+    int s = int(seed);
+    if (s == 0) {
+        s = 305420679;
+    }
+    k = s / 127773;
+    s = 16807 * (s - k * 127773) - 2836 * k;
+    if (s < 0) {
+        s += 2147483647;
+    }
+    seed = uint(s);
+    return float(seed % uint(65536)) / 65535.0;
+}
+
+uint hash(uint x) {
+    x = ((x >> uint(16)) ^ x) * uint(73244475);
+    x = ((x >> uint(16)) ^ x) * uint(73244475);
+    x = (x >> uint(16)) ^ x;
+    return x;
+}
+
+void start() {
+    uint base_number = NUMBER;
```

Comment:
There's also a few variables here that need renames to be consistent with our naming convention.